lib/run.js

```diff
@@ -137,80 +139,83 @@ function _run (opts, cb, tracker) {
     counter = 0
     bytes = 0
 
-    tracker.emit('tick')
+    tracker.emit('tick', totalCompletedRequests)
   }
 
-  function tickInterval () {
-    if (!stop) {
-      updateStat()
-
-      return true
+  function stopProcess () {
+    updateStat()
+
+    const result = {
+      title: opts.title,
+      url: opts.url,
+      socketPath: opts.socketPath,
+      requests: addPercentiles(requests, histAsObj(requests, totalCompletedRequests)),
+      latency: addPercentiles(latencies, histAsObj(latencies)),
+      throughput: addPercentiles(throughput, histAsObj(throughput, totalBytes)),
+      errors: errors,
+      timeouts: timeouts,
+      duration: Math.round((Date.now() - startTime) / 10) / 100,
+      start: new Date(startTime),
+      finish: new Date(),
+      connections: opts.connections,
+      pipelining: opts.pipelining,
+      'non2xx': statusCodes[0] + statusCodes[2] + statusCodes[3] + statusCodes[4]
     }
+    result.requests.sent = totalRequests
+    statusCodes.forEach((code, index) => { result[(index + 1) + 'xx'] = code })
+    if (result.throughput.min === Number.MAX_SAFE_INTEGER) result.throughput.min = 0
+    if (result.latency.min === Number.MAX_SAFE_INTEGER) result.latency.min = 0
+
+    tracker.emit('done', result)
+    if (!opts.forever) cb(null, result)
+
+    // the restart function
+    setImmediate(() => {
```

Comment:
Can you move this closure to a top level function?
Why are you calling it with `setImmediate`?